app/controllers/salt_controller.rb

```diff
@@ -0,0 +1,29 @@
+require "velum/salt"
+require "velum/salt_api"
+# HealthController holds methods for checking the readiness of the application.
+class SaltController < ApplicationController
+	include Velum::SaltApi
+ 	skip_before_action :redirect_to_setup
+
+	def update
+
+    minions = perform_request(endpoint: "/minions", method: "get")
+    ret = JSON.parse(minions.body)["return"]
+
+    for minion in ret do
+      minion.each do | key, value |
+        if value["tx_update_reboot_needed"]
+        	Minion.where(minion_id: key).update_all( highstate: Minion.highstates[:pending] )
```

Comment:
I am not sure - `Minion.highstates` is an `enum` - I have no idea how ruby deals with them